### ISSUE TITLE
Recording inserts too many notes.

### ISSUE THREAD

**Issue by theGreatWhiteShark** [hydrogen-music/hydrogen#1224]
Hydrogen version *          : 7bcfb5922886bbc0c2459a6dbca30ae706b1d50b
Operating system + version  :  Linux
Audio driver + version      :  JACK

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

When recording one note over another note two instead of one notes will be inserted.

Steps to produce:

1. Open a new song
2. Press record button
3. Press play button
4. Press <kbd>x</kbd> (or any other button associated with Hydrogen's virtual keyboard in its input mode
5. In the looped recording try to hit the same spot again

When checking the corresponding (saved) .h2song file after 4. one note got inserted and after 5. three notes have been inserted.

Every time you hit the same spot again, two additional notes will be inserted.

**Comment by oddtime** [comment]
In` src/gui/src/HydrogenApp.cpp` the action is repeated 2 times, the comment suggests that each time the note should be deleted and added again, which doesn't happen. Apparently changing `rounds=1` solves the problem.

```
	// midi notes
	while(!pQueue->m_addMidiNoteVector.empty()){

		int rounds = 1;
		if(pQueue->m_addMidiNoteVector[0].b_noteExist) { // run twice, delete old note and add new note. this let the undo stack consistent 
			rounds = 2;
		}
```

**Comment by oddtime** [comment]
A brief reflection. Thinking about the use of this feature (enter a pattern), probably the new recorded note should replace the old
(why should one overdubbing the same note? there is chance).
Unless the notes are input in instrument mode with different pitches...

Otherwise, it is sufficient to change those lines in `src/gui/src/HydrogenApp.cpp`

**Comment by oddtime** [comment]
I have opened a related pull req #1224

**Comment by theGreatWhiteShark** [comment]
> Thinking about the use of this feature (enter a pattern), probably the new recorded note should replace the old
(why should the user overdub the same note? there is chance that it is done unintentionally and not to "boost" that note).

Yes. Especially since starting with version 1.1 the user is not able to place multiple notes with the same pitch at the same position via mouse or keyboard editing anymore.

**Comment by theGreatWhiteShark** [comment]
Fixed in #1247
